dpcpp/CMakeLists.txt

```diff
@@ -67,7 +67,7 @@ target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_COMPILER_FLAGS}")
 # Note: add MKL as PRIVATE not PUBLIC (MKL example shows) to avoid propagating
 # find_package(MKL) everywhere when linking ginkgo (see the MKL example
 # https://software.intel.com/content/www/us/en/develop/documentation/onemkl-windows-developer-guide/top/getting-started/cmake-config-for-onemkl.html)
-target_compile_features(ginkgo_dpcpp PRIVATE cxx_std_17)
+target_compile_features(ginkgo_dpcpp PUBLIC cxx_std_17)
```

Comment:
I think it is safer to make sure all pieces of Ginkgo use the same compiler and standard version, otherwise we may get subtle binary incompatibilities.